xmodule/docs/decisions/0004-restructure-video-block.rst

```diff
@@ -0,0 +1,101 @@
+
+Accessibility is a key focus area, and the new player aims to be compliant with the latest WCAG guidelines. This includes providing Audio Descriptions, focus state visibility, and other accessibility features, which are currently either lacking or not up to the mark in the existing system. The new architecture aims to rectify this by integrating these features into the core of the new Video Player.
+
+In summary, the context for this Architecture Decision Record (ADR) is introduction of the new Video XBlock aims to address multiple pain points, ranging from user experience and accessibility to extensibility and modernization of the tech stack. It represents a significant step forward in aligning the platform with modern web standards and user expectations, while also laying the groundwork for future enhancements.
+
+
+Decision
+--------
+
+1. **Move out current Video XBlock from xmodules and place it in a separate repository**
+
+   We have decided to extract the current Video XBlock from the xmodules directory and relocate it into its own dedicated repository within the github.com/openedx organization. This decision is driven by the need for improved modularity, maintainability, and the facilitation of continuous integration and deployment practices.
+   * A new repository will be created under the Open edX GitHub organization.
+   * The repository will adhere to Open edX's best practices, including code quality, testing, and documentation.
+   * The new XBlock will be hosted within the Open edX organization on GitHub, ensuring community involvement and adherence to Open edX standards.
+   * The proposed architecture for installing and using new xblock is the following:
+
+     .. image:: ./Architecture.png
+
+2. **React.js and Video.js Integration**
+   * React.js will be used to build a responsive, accessible frontend for a new Xblock as proposed in https://github.com/openedx/XBlock/issues/635 and prototyped in https://github.com/openedx/XBlock/issues/634.
+   * Video.js will be integrated for video playback, following the official guidelines (`Video.js Docs <https://docs.videojs.com/>`_).
+   * An NPM package will be created for the frontend, and it will be versioned and published (`Issue 635 <https://github.com/openedx/XBlock/issues/635>`_).
+
+3. **Feature Toggle in edx-platform**
+   * A feature toggle will be introduced using Django Waffle flags.
+   * This will allow for a manual rollout and the ability to quickly revert to the old player if issues arise.
+
+4. **Integration into Legacy Interface and Course Authoring**
+   * The new XBlock will be embedded in iframes in the legacy interface.
+   * In the Course Authoring MFE, the XBlock will be integrated natively using React.js components.
```

Comment:
Is there any precedent from other XBlocks for what you're doing here? If so, please cite. This seems like an area that could add a lot of complexity.

Reply:
I believe ORA2 team is doing something pretty similar, I will investigate and provide more info on this ADR.

Besides this, the approach suggested by @kdmccormick in https://github.com/openedx/xBlock/issues/634 is compatible with this decision. However, it also differs in the area of where to render the required front-end in the LMS, whether inside or outside of the iframe.
Eventually, the ReactJS component will be packaged in an NPM package, awaiting props to render the Video Player. These props can be passed to the component either from the VideoBlock backend or directly from the new Course/Library Authoring editors.

I will note this in ADR